framework/decode/vulkan_replay_consumer_base.cpp

```diff
@@ -8481,25 +8478,42 @@ void VulkanReplayConsumerBase::OverrideCmdTraceRaysKHR(
 
         if (!(valid_group_handles && valid_sbt_alignment))
         {
-            // TODO: remove TODO/warning when issue #1526 is solved
-            GFXRECON_LOG_WARNING_ONCE("OverrideCmdTraceRaysKHR: invalid shader-binding-table (size, alignment, handles")
+            if (!valid_sbt_alignment)
+            {
+                // TODO: remove TODO/warning when issue #1526 is solved
+                GFXRECON_LOG_WARNING_ONCE("OverrideCmdTraceRaysKHR: invalid shader-binding-table (handle-size and/or "
+                                          "alignments mismatch) -> TODO: run SBT re-assembly");
 
-            // TODO: run Repl8cer! (create linear hashmap, run compute-shader, profit)
+                // TODO: create shadow-SBT-buffer, remap addresses to that
+            }
+            else
+            {
+                // in-place remap: capture-addresses -> replay-addresses
+                address_remap(in_pRaygenShaderBindingTable);
+                address_remap(in_pMissShaderBindingTable);
+                address_remap(in_pHitShaderBindingTable);
+                address_remap(in_pCallableShaderBindingTable);
+            }
+
+            // TODO: run Repl8cer! (create linear hashmap (x), run compute-shader (), profit ())
             util::linear_hashmap<graphics::shader_group_handle_t, graphics::shader_group_handle_t> hashmap;
             for (const auto& [lhs, rhs] : shader_group_handles)
             {
                 hashmap.put(lhs, rhs);
             }
         }
-
-        func(commandBuffer,
-             in_pRaygenShaderBindingTable,
-             in_pMissShaderBindingTable,
-             in_pHitShaderBindingTable,
-             in_pCallableShaderBindingTable,
-             width,
-             height,
-             depth);
+        else
+        {
+            // tmp: guard execution, this would crash otherwise
+            func(commandBuffer,
```

Comment:
Isn't this call still needed in the above `if`?

Reply:
yes, absolutely. we will always want to call this function in the end.

the branching above should just set the stage for it (optionally adapting addresses, fixing binding-tables, etc.).

currently the condition omits to call `vkCmdTraceRays` in case we know it will crash (because the sbt is not valid and we did not yet fix it) .
this was mainly for me to find out how far we get after that point, will remove the `else` clause

Reply:
`vkCmdTraceRays` not skipped anymore